# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../udev/legacy_sus.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

usleep() was deprecated by SUSv3 and newer LIBCs are starting to enforce that
deprecation.

--- ./udev/udev.h.orig	2010-09-25 18:37:03.384000255 +0200
+++ ./udev/udev.h	2010-09-25 18:42:20.348000252 +0200
@@ -52,6 +52,13 @@
 	closelog();
 }
 
+#define usleep(x)	xx_usleep(x)
+static inline void xx_usleep(unsigned k)
+{
+	const struct timespec t = { k/1000000UL, (k%1000000UL)*1000UL };
+	nanosleep(&t, NULL);
+}
+
 struct udev_event {
 	struct udev_list_node node;
 	struct udev *udev;
